ivy_tests/test_ivy/test_frontends/test_jax/test_jax_src_tree_util.py

```diff
@@ -1 +1,66 @@
-# global
+# local
+import ivy
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+from ivy.functional.frontends.jax._src.tree_util import tree_leaves, tree_map
+import hypothesis.strategies as st
+
+
+# Define a function to square each leaf node
+def square(x):
+    if isinstance(x, ivy.Array):
+        return ivy.square(x)
+    else:
+        return x**2
+
+
+def leaf_strategy():
+    return st.lists(st.integers(1, 10)).map(ivy.array)
+
+
+def tree_strategy(max_depth=2):
+    if max_depth == 0:
+        return leaf_strategy()
+    else:
+        return st.dictionaries(
+            keys=st.one_of(
+                *[
+                    st.text(
+                        alphabet=st.characters(min_codepoint=97, max_codepoint=122),
+                        min_size=1,
+                        max_size=1,
+                    ).filter(lambda x: x not in used_keys)
+                    for used_keys in [set()]
+                ]
+            ),
+            values=st.one_of(leaf_strategy(), tree_strategy(max_depth - 1)),
+            min_size=1,
+            max_size=10,
+        )
+
+
+@st.composite
+def tree_dict_strategy(draw):
+    return draw(tree_strategy())

```

Comment:
Great! My only suggestion is you could define `tree_strategy` directly as a composite instead of using the extra `tree_dict_strategy`.

Reply:
I have already tried but it gives me following error because tree_strategy is recursive procedure :

`unsupported operand type(s) for ** or pow(): 'MappedSearchStrategy' and 'int'`